src/sbt-slicer.cpp

```diff
@@ -1,156 +1,141 @@
-                   "of the given function, but the instructions that\n"
-                   "follow the call. I.e. the call is used just to mark\n"
-                   "the instruction.\n"
-                   "E.g. for 'crit' being set as the criterion, slicing critera "
-                   "are all instructions that follow any call of 'crit'.\n"),
-                   llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
-
-llvm::cl::opt<bool> memsafety("memsafety",
-    llvm::cl::desc("Assume that slicing criteria are potentially memory-unsafe "
-                   "instructions. This option implies criteria-are-next-instr, "
-                   "i.e., the slicing criteria are \"only marked\" "
-                   "by the given slicing criteria."),
-                   llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+using AnnotationOptsT =
+        dg::debug::LLVMDGAssemblyAnnotationWriter::AnnotationOptsT;
+
+llvm::cl::opt<bool> enable_debug(
+        "dbg", llvm::cl::desc("Enable debugging messages (default=false)."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> should_verify_module(
+        "dont-verify", llvm::cl::desc("Verify sliced module (default=true)."),
+        llvm::cl::init(true), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> abort_on_threads(
+        "abort-on-threads",
+        llvm::cl::desc("Abort when threads are present (default=true)."),
+        llvm::cl::init(true), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool>
+        abort_on_unsupp("abort-on-unsupported",
+                        llvm::cl::desc("Abort when some unsupported feature is "
+                                       "present (default=true)."),
+                        llvm::cl::init(true), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> remove_unused_only(
+        "remove-unused-only",
+        llvm::cl::desc("Only remove unused parts of module (default=false)."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> statistics(
+        "statistics",
+        llvm::cl::desc("Print statistics about slicing (default=false)."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool>
+        dump_dg("dump-dg",
+                llvm::cl::desc("Dump dependence graph to dot (default=false)."),
+                llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> dump_dg_only(
+        "dump-dg-only",
+        llvm::cl::desc("Only dump dependence graph to dot,"
+                       " do not slice the module (default=false)."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> dump_bb_only(
+        "dump-bb-only",
+        llvm::cl::desc("Only dump basic blocks of dependence graph to dot"
+                       " (default=false)."),
+        llvm::cl::init(false), llvm::cl::cat(SlicingOpts));
+
+llvm::cl::opt<bool> criteria_are_next_instr(
```

Comment:
I wonder if we shouldn't move this option (and maybe some others, like `criteria-are-next-instr` and `cutoff-diverging`) with their implementation to `llvmdgslicer`?

Reply:
It would simplify any further syncing.

Reply:
If you give a full list of such options, I'll gladly prepare such PR.

Reply:
I guess just these two. Btw. the code that runs cutoff-diverging should be probably made a method of `Slicer` rather than creating a new function in `llvmdgslicer`.

Reply:
Resolved in ac52ba3.